Source/DataSources/GeometryVisualizer.js

```diff
@@ -143,15 +148,25 @@ define([
         this._removedObjects = new AssociativeArray();
         this._changedObjects = new AssociativeArray();
 
-        this._outlineBatch = new StaticOutlineGeometryBatch(primitives, scene);
-        this._closedColorBatch = new StaticGeometryColorBatch(primitives, type.perInstanceColorAppearanceType, true);
-        this._closedMaterialBatch = new StaticGeometryPerMaterialBatch(primitives, type.materialAppearanceType, true);
-        this._openColorBatch = new StaticGeometryColorBatch(primitives, type.perInstanceColorAppearanceType, false);
-        this._openMaterialBatch = new StaticGeometryPerMaterialBatch(primitives, type.materialAppearanceType, false);
+        var numberOfShadowModes = 4;
+        this._outlineBatches = new Array(numberOfShadowModes);
+        this._closedColorBatches = new Array(numberOfShadowModes);
+        this._closedMaterialBatches = new Array(numberOfShadowModes);
+        this._openColorBatches = new Array(numberOfShadowModes);
+        this._openMaterialBatches = new Array(numberOfShadowModes);
+
+        for (var i = 0; i < numberOfShadowModes; ++i) {
```

Comment:
I'd like @shunter to chime in here, but I believe we should be using `length` as a name instead of `NUMBER_OF_SHADOW_MODES`.